README.md

```diff
@@ -203,14 +203,34 @@ Currently modis_tools only supports downloading of hdf file type.
 
 ## Development and Testing
 
+### Pre-commit hooks
+
+- The developers use [pre-commit](https://pre-commit.com/) hooks to format code before committing. pre-commit can be installed with:
+  ```bash
+  pip install pre-commit
+  ```
+  and should be run before committing with:
+  ```bash
+  pre-commit run --all-files 
+  ```
+
 ### Setting up a development environment
 
-- To install all production dependencies, run:
+- The developers use [poetry](https://python-poetry.org/) to manage virtual environments and dependencies. Included in root is a `pyproject.toml` and `poetry.lock`. To quickly setup a virtual environment locally, from root simply run:
+
+  ```bash
+  poetry shell
+  poetry update
+  ```
+
+- Alternatively, To install all production dependencies, run:
```

Comment:
I don't think the move is to add poetry here. We deploy this as a pip installable package using requirements and setup.py. Adding anything related to poetry conflates access strategy. Specifically, the pyproject.toml created here (despite not being specifically a poetry file) will cause drift with setup.py. I recommend we remove the pyproj